include/api.php

```diff
@@ -565,9 +565,7 @@ function api_get_user(&$a, $contact_id = Null, $type = "json"){
 
 			//AND `allow_cid`='' AND `allow_gid`='' AND `deny_cid`='' AND `deny_gid`=''",
 			// count public wall messages
-			$r = q("SELECT count(*) as `count` FROM `item`
-					WHERE  `uid` = %d
-					AND `type`='wall'",
+			$r = q("SELECT COUNT(*) as `count` FROM `item` WHERE `uid` = %d AND wall",
```

Comment:
When the field "type" has the value "wall" then the field "wall" is true. Yeah, that is confusing, I know. I want to change all queries to the item table to one of these two fields. So maybe then we can get rid of some indexes.